docs/reference/analysis/tokenizers/edgengram-tokenizer.asciidoc

```diff
@@ -1,80 +1,324 @@
-                },
-                "tokenizer" : {
-                    "my_edge_ngram_tokenizer" : {
-                        "type" : "edgeNGram",
-                        "min_gram" : "2",
-                        "max_gram" : "5",
-                        "token_chars": [ "letter", "digit" ]
-                    }
-                }
-            }
+      "token": "Fo",
+      "start_offset": 8,
+      "end_offset": 10,
+      "type": "word",
+      "position": 4
+    },
+    {
+      "token": "Fox",
+      "start_offset": 8,
+      "end_offset": 11,
+      "type": "word",
+      "position": 5
+    },
+    {
+      "token": "Foxe",
+      "start_offset": 8,
+      "end_offset": 12,
+      "type": "word",
+      "position": 6
+    },
+    {
+      "token": "Foxes",
+      "start_offset": 8,
+      "end_offset": 13,
+      "type": "word",
+      "position": 7
+    }
+  ]
+}
+----------------------------
+// TESTRESPONSE
+
+/////////////////////
+
+The above example produces the following terms:
+
+[source,text]
+---------------------------
+[ Qu, Qui, Quic, Quick, Fo, Fox, Foxe, Foxes ]
+---------------------------
+
+Usually we recommend using the same `analyzer` at index time and at search
+time. In the case of the `edge_ngram` tokenizer, the advice is different.  It
+only makes sense to use the `edge_ngram` tokenizer at index time, to ensure
+that partial words are available for matching in the index.  At search time,
+just search for the terms the user has typed in, for instance: `Quick Fo`.
+
+Below is an example of how to set up a field for _search-as-you-type_:
+
+[source,js]
+-----------------------------------
+PUT my_index
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "autocomplete": {
+          "tokenizer": "autocomplete",
+          "filter": [
+            "lowercase"
+          ]
+        },
+        "autocomplete_search": {
+          "tokenizer": "lowercase"
+        }
+      },
+      "tokenizer": {
+        "autocomplete": {
+          "type": "edge_ngram",
+          "min_gram": 2,
+          "max_gram": 10,
+          "token_chars": [
+            "letter"
+          ]
         }
-    }'
+      }
+    }
+  },
+  "mappings": {
+    "doc": {
+      "properties": {
+        "title": {
+          "type": "text",
+          "analyzer": "autocomplete",
+          "search_analyzer": "autocomplete_search"
+        }
+      }
+    }
+  }
+}
 
-    curl 'localhost:9200/test/_analyze?pretty=1&analyzer=my_edge_ngram_analyzer' -d 'FC Schalke 04'
-    # FC, Sc, Sch, Scha, Schal, 04
---------------------------------------------------
+PUT my_index/doc/1
+{
+  "title": "Quick Foxes" <1>
+}
 
-[float]
-==== `side` deprecated
+POST my_index/_refresh
+
+GET my_index/_search
+{
+  "query": {
+    "match": {
+      "title": {
+        "query": "Quick Fo", <2>
+        "operator": "and"
+      }
+    }
+  }
+}
+-----------------------------------
+// CONSOLE
+// TEST[skip:The took value will change on every request]
```

Comment:
s/"took": 123/"took": "$body.took"/ is the way I've done it. Might be the
best way but it works.
On May 15, 2016 8:07 AM, "Clinton Gormley" <notifications@github.com> wrote:

> In docs/reference/analysis/tokenizers/edgengram-tokenizer.asciidoc
> <https://github.com/elastic/elasticsearch/pull/18356#discussion_r63290592>
> :
>
> > +POST my_index/_refresh
> > +
> > +GET my_index/_search
> > +{
> > +  "query": {
> > +    "match": {
> > +      "title": {
> > +        "query": "Quick Fo", <2>
> > +        "operator": "and"
> > +      }
> > +    }
> > +  }
> > +}
> > +-----------------------------------
> > +// CONSOLE
> > +// TEST[skip:The took value will change on every request]
>
> This GET request will return a different took value every time. I tried
> just removing it with with TEST[s/"took".*"//] but that didn't work.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/18356/files/e79d5a85afc6fde119c7ab58f04a1076a1456391#r63290592>
>